Dockerfiles/Dockerfile

```diff
@@ -1,8 +1,11 @@
 # Build the manager binary
 ARG GOLANG_VERSION=1.22
 
+ARG BUILDPLATFORM
+ARG TARGETPLATFORM
+
 ################################################################################
-FROM registry.access.redhat.com/ubi8/toolbox as manifests
+FROM --platform=$BUILDPLATFORM registry.access.redhat.com/ubi8/toolbox as manifests
```

Comment:
I think this makes sense to run on the build platform if we use cross-compilation as it will be quicker. However, for me it works also without this change if I run with `--platform linux/arm64` on my Fedora x86_64 because Podman/Docker will then use emulation, so it takes longer but I cannot see any go binary mismatch errors. Then the go binary of the target platform is emulated.

The whole command was:
```bash
podman build --platform linux/arm64 -f Dockerfiles/Dockerfile   \ 
--build-arg CGO_ENABLED=0  \
 -t odh-operator-test:latest .
```
Bear in mind I'm on Fedora x86_64, so I targeted image to `linux/arm64`

Reply:
@grdryn to clarify - I meant that I ran the command without your changes from this PR and I cannot see any go binary mismatch errors. But you said in the description:

> The main problem is that the default is to use the TARGETPLATFORM for
all stages in an image build, which means that, for example, the go
commands that are run in the builder stage will be using an x86_64
go binary, which essentially doesn't work on aarch64.

But vice versa to your use case (running aarch64 binaries using emulation on x86_64) it is working for me without any change from this PR.

Reply:
> But vice versa to your use case (running aarch64 binaries using emulation on x86_64) it is working for me without any change from this PR.

Probably, you have binfmt_misc configured to run via qemu on linux.

Reply:
@ykaliuta @grdryn yes, that is the standard emulation which should come automatically with every installation AFAIK.
See: https://podman-desktop.io/docs/podman/rosetta